### DISCUSSION TITLE
ISO oracle integration information

### DISCUSSION THREAD

**Discussion by tgrecojs** [Agoric/agoric-sdk#7043] [Q&A] [answered]
With the release of inter-protocol's vaults continues to approach, I'm seeing more and more issues being crossed off regarding oracle integration, however, I'm still unsure of what the initial implementation is planned to look like. 

Can someone aid me in finding out more information around this? A few weeks back I heard something about a chainlink solution that simply staking was providing so if this is the case, then it would be really helpful if someone could help me point me in the direction for getting more concrete info regarding this solution.  

[Dapp-oracle](https://github.com/Agoric/dapp-oracle) is now marked as public archive. It's worth mentioning that I really appreciate seeing the different set of components that this example dapp is comprised of, and so I'm thankful that it will remain on the org page. 

That said, apparently the solution put forth in dapp-oracle wasn't a viable one for a production application, so information as to why exactly this is the case would be greatly appreciated as well.

Thanks! 

🟥 ❤️ 🧊 🍵

**Comment by turadg** [comment, marked as answer]
https://github.com/Agoric/agoric-sdk/tree/master/packages/inter-protocol/src/price has documentation on the price oracles for Inter Protocol.

dapp-oracle was never designed for production:
> Note that using this in production is not recommended: you will have to ensure your ag-solo as always available, or your contracts will not be able to query it. Running a robust oracle is a detailed and time-consuming endeavour, and so we recommend instead that you use the Chainlink oracles already provided as a service on the Agoric chain.

So that oracles can safely lose connectivity, they are implemented as clients to the Smart Wallet. Their rights to push prices are held in their invitation makers received from an invitation to be an oracle. (Analagous to how an Economic Committee member's right to vote.)

**Reply by tgrecojs** [reply]
Thanks for the info @turadg . 

I've looked through inter's src/price directory and it's been helpful. So has the info surrounding the smart wallet's role in the process of getting price data, which we were able to discuss a bit today during office hours. (cc @dckc) 

I'm primarily curious about the oracle technology that simply staking has been helping create, which I presume will be integrated with inter protocol in the near future. Chainlink has a variety of services that agoric dapps could benefit from integrating with, so was wondering about what that might look like.

It's not a blocker though. The information that I have at this point in time is sufficient 👍 🙂

**Comment by otoole-brendan** [comment]
@tgrecojs We'd be happy to share more detailed information in the coming weeks - we're currently focused on the Vaults release and finalizing the underlying Oracle functional details and documentation.

**Comment by dckc** [comment]
p.s. note community forum discussion of oracle operations in May:

 - [Inter Protocol Vaults: Contract Implementations](https://community.agoric.com/t/inter-protocol-vaults-contract-implementations/261), especially
   - [Simply Staking's comments](https://community.agoric.com/t/inter-protocol-vaults-contract-implementations/261/29)
